tests/unit/service_test.py

```diff
@@ -254,6 +254,15 @@ def test_recreate_container(self, _):
         new_container.start.assert_called_once_with()
         mock_container.remove.assert_called_once_with()
 
+    @mock.patch('compose.service.Container', autospec=True)
+    def test_recreate_container_with_timeout(self, _):
+        mock_container = mock.create_autospec(Container)
+        service = Service('foo', client=self.mock_client, image='someimage')
+        service.image = lambda: {'Id': 'abc123'}
```

Comment:
The unit-under-test (in this case `Service`) really shouldn't be modified, otherwise you risk having the unit behave different, and the test not reflect normal operation.

Could this be provided as a return value from mock_client instead?

```
mock_client.inspect_image.return_value = {'Id': ...}
```